### PR TITLE
Doc code and typos

### PR DESCRIPTION
**Pull request by eumiro**
Fixed some typos in the doc. Modernized the example code snippets.

Then generated everything with `make html`, copied to the `pytables.github.com` repo, but the diff is much larger than expected due to some IE/class/… changes. The new version is also `3.6.2.dev0`, so probably we should run this update when the new release is ready.

### CODE REVIEW

File: doc/scripts/pickletrouble.py [hunk 1]
```diff
@@ -4,24 +4,25 @@
...
+    print(h5f.root._v_attrs.obj.foo)
+    # Let us inspect the object to see what is happening.
+    print(repr(h5f.root._v_attrs.obj))
+    # Maybe unpickling the string will yield more information:
+    import pickle
+
+    pickle.loads(h5f.root._v_attrs.obj)
+    # So the problem was not in the stored object,
+    # but in the *environment* where it was restored.
+    h5f.close()
```
Comment (avalentino) [inline doc/scripts/pickletrouble.py:28]: no need to close if one uses the with statement

File: doc/source/usersguide/tutorials.rst [hunk 2]
```diff
@@ -1227,18 +1227,18 @@ can be passed to this method::
...
 
             ########### Detectable errors start here. Play with them!
-            particle['pressure'] = array(i*arange(2*3)).reshape((2,4))  # Incorrect
-            #particle['pressure'] = array(i*arange(2*3)).reshape((2,3)) # Correct
+            particle['pressure'] = np.array(i * np.arange(2 * 3)).reshape((2, 4))  # Incorrect
+            #particle['pressure'] = np.array(i * np.arange(2 * 3)).reshape((2, 3)) # Correct
             ########### End of errors
 
-            particle['temperature'] = (i**2)     # Broadcasting
+            particle['temperature'] = i ** 2     # Broadcasting
```
Comment (avalentino) [inline doc/source/usersguide/tutorials.rst:1242]: Personally I'm not used to see spaces around the ``**`` operator but it is OK

**Review by avalentino — APPROVED** [review]
Just a couple of minor comments

**Comment by avalentino** [comment]
> The new version is also 3.6.2.dev0, so probably we should run this update when the new release is ready.

Probably yes but in the end there are not too many differences between 3.6.1. and 3.6.2.dev0 so also updating now would be fine to me.

File: doc/source/usersguide/tutorials.rst (hunk 2 above)
Comment (eumiro) [reply doc/source/usersguide/tutorials.rst:1242]: Would we consider introducing some sort of automatic formatter for the code, such as `black`? I believe `black` puts spaces there, but with a formatter (and a `pre-commit` rule that would apply all those changes to any commit) there would be no need to take special care of it.

File: doc/scripts/pickletrouble.py (hunk 1 above)
Comment (eumiro) [reply doc/scripts/pickletrouble.py]: Thank you, fixed and repushed.

File: doc/source/usersguide/tutorials.rst (hunk 2 above)
Comment (avalentino) [reply doc/source/usersguide/tutorials.rst:1242]: Honestly I'm not a big fan of pre-commit hooks the modify the code.
I have always checked the code with [flake8](https://flake8.pycqa.org) if we want to switch to black I'm fine with it but I would prefer to to use this tools offline.

Comment (eumiro) [reply doc/source/usersguide/tutorials.rst:1242]: They are meant to check whether the code has been formatted/linted properly and ask the contributor to do so. `flake8` is one of the options, of course.

**Comment by eumiro** [comment]
Pushed to https://github.com/PyTables/pytables.github.com and the new version is live. Thank you!

**Pull request merged**